File: backend/src/main/java/net/zorphy/backend/main/dto/game/GameFilters.java
package net.zorphy.backend.main.dto.game;


import java.time.Duration;
import java.time.Instant;
import java.util.List;
import java.util.UUID;

public record GameFilters(
        String text,
        Instant dateFrom,
        Instant dateTo,
        Duration minDuration,
        Duration maxDuration,
        List<GameType> gameTypes,
        List<UUID> players
) {
}

File: frontend/src/app/main/dto/games/GameFilters.ts
import {GameType} from "./GameType";

export interface GameFilters {
    text: string | null;
    dateFrom: string | null,
    dateTo: string | null,
    minDuration: string | null,
    maxDuration: string | null,
    gameTypes: GameType[] | null,
    players: string[] | null
}